build.sbt

```diff
@@ -116,6 +122,35 @@ lazy val codegenJava =
     .settings(Defaults.itSettings)
     .settings(name := "akkaserverless-codegen-java", testFrameworks += new TestFramework("munit.Framework"))
     .settings(Dependencies.codegenJava)
+    .settings(crossScalaVersions := Dependencies.ScalaVersionForCodegen,
+              scalaVersion := Dependencies.ScalaVersionForCodegen.head)
+
+lazy val codegenSbtPlugin = project
+  .in(file("codegen/sbt-plugin"))
+  .enablePlugins(SbtPlugin)
+  .dependsOn(codegenJava)
+  .settings(
+    (publish / skip) := true,
+    name := "akkaserverless-codegen-sbt-plugin",
+    Compile / javacOptions ++= Seq("-encoding", "UTF-8", "-source", "11", "-target", "11")
+  )
+  .settings(crossScalaVersions := Dependencies.ScalaVersionForCodegen,
+            scalaVersion := Dependencies.ScalaVersionForCodegen.head)
```

Comment:
Plugin needs to use Scala 2.12 and therefore `codegenJava` and `codegenCore` as well.

Note that for the maven plugin that will call the codegen it doesn't matter if it's 2.12 or 2.13